library/core/src/panicking.rs

```diff
@@ -245,7 +247,14 @@ pub fn assert_matches_failed<T: fmt::Debug + ?Sized>(
             f.write_str(self.0)
         }
     }
-    assert_failed_inner(AssertKind::Match, &left, &Pattern(right), args);
+    assert_failed_inner(
+        AssertKind::Match,
+        &left,
+        &Pattern(right),
+        stringify!(&left),
+        stringify!(&right),
```

Comment:
This will create the literal string `"&right"`, which is probably not what you want. I think `right` is already stringified here, but you need to stringify `left` in the `assert_matches!` macro.

Reply:
thx, fixed